### PR TITLE
Report created RTE daemonsets in operator status and fix status sync issue

### PR DESCRIPTION
**Pull request by shajmakh**
Complementary unit tests coverage for #1018 in addition to fixing other deficient behaviors; see the commits messages for more info.

### CODE REVIEW

File: controllers/numaresourcesoperator_controller.go [hunk 1]
```diff
@@ -769,7 +769,7 @@ func daemonsetUpdater(mcpName string, gdm *rtestate.GeneratedDesiredManifest) er
 }
 
 func isDaemonSetReady(ds *appsv1.DaemonSet) bool {
-	ok := (ds.Status.DesiredNumberScheduled > 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady)
+	ok := (ds.Status.DesiredNumberScheduled > 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady) || ds.Status.DesiredNumberScheduled == 0
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller.go:772]: please split this complex logic in some simpler `if` blocks, adding subfunctions if it helps

File: controllers/numaresourcesoperator_controller_test.go [hunk 2]
```diff
@@ -261,7 +261,7 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 
 			secondLoopResult, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
 			Expect(err).ToNot(HaveOccurred())
-			Expect(secondLoopResult).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+			Expect(secondLoopResult).To(Equal(reconcile.Result{RequeueAfter: 0}))
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller_test.go:262]: let's add comments explaining what we expect and why (here and in the following code)

**Review by ffromani — COMMENTED** [review]
nice work

File: controllers/numaresourcesoperator_controller_test.go (hunk 2 above)
Comment (shajmakh) [reply controllers/numaresourcesoperator_controller_test.go]: Worth it indeed, added in all relevant places. Thanks.

File: controllers/numaresourcesoperator_controller_test.go [hunk 3]
```diff
@@ -446,7 +448,136 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
...
+					PoolName: &mcp1Name,
+					Config:   conf1,
+				}
+				ng2 := nropv1.NodeGroup{
+					PoolName: &mcp2Name,
+					Config:   conf2,
+				}
+
+				mcp1 := testobjs.NewMachineConfigPool(mcp1Name, label1, &metav1.LabelSelector{MatchLabels: label1}, &metav1.LabelSelector{MatchLabels: label1})
+				mcp2 := testobjs.NewMachineConfigPool(mcp2Name, label2, &metav1.LabelSelector{MatchLabels: label2}, &metav1.LabelSelector{MatchLabels: label2})
```
Comment (ffromani) [inline controllers/numaresourcesoperator_controller_test.go:498]: nit: variable definition can be as close as possible to the first usage to minimize scope (and thus cognitive load)

**Review by ffromani — COMMENTED** [review]
/approve

File: controllers/numaresourcesoperator_controller_test.go (hunk 3 above)
Comment (ffromani) [reply controllers/numaresourcesoperator_controller_test.go:498]: please do NOT resubmit only to fix this

**Pull request merged**